src/tensor_conversion.cpp

```diff
@@ -464,15 +506,47 @@ static Status convertNativeFileFormatRequestTensorToOVTensor(const TensorType& s
     return StatusCode::OK;
 }
 
+static Status convertStringRequestFromBufferToOVTensor2D(const tensorflow::TensorProto& src, ov::Tensor& tensor, const std::string* buffer) {
+    return StatusCode::NOT_IMPLEMENTED;
+}
+
+static Status convertStringRequestFromBufferToOVTensor2D(const ::KFSRequest::InferInputTensor& src, ov::Tensor& tensor, const std::string* buffer) {
+    int batchSize = 0;
+    size_t offset = 0;
+    if (buffer->size() < sizeof(uint32_t)) {
+        return StatusCode::INVALID_STRING_INPUT;
+    }
+    size_t maxStringLength = 0;
+    while (offset < buffer->size()) {
+        uint64_t inputSize = *((int32_t*)(buffer->data() + offset));
+        offset += (sizeof(uint32_t) + inputSize);
+        maxStringLength = std::max(maxStringLength, inputSize);
+        batchSize++;
+    }
+    if (offset != buffer->size()) {
+        return StatusCode::INVALID_STRING_INPUT;
+    }
+    size_t width = maxStringLength + 1;
+    offset = 0;
+    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<uint64_t>(batchSize), width});
+    for (int i = 0; i < batchSize; i++) {
+        int32_t inputSize = *((int32_t*)(buffer->data() + offset));
+        offset += sizeof(uint32_t);
+        std::memcpy(tensor.data<unsigned char>() + i * width, reinterpret_cast<const unsigned char*>(buffer->data() + offset), inputSize);
+        tensor.data<unsigned char>()[i * width + inputSize] = 0;
```

Comment:
In non buffer scenario we fill remaining data with zeros, we should do the same in both cases